File: Trabalho1/Trabalho1/src/Linha.cpp


#include <cstdlib>

#include "Linha.h"
#include "AABB.h"

Linha::Linha()
{
    //this->aabb.reset();
    this->aabb = nullptr;
    this->x1 = 0;
    this->y1 = 0;
    this->x2 = 0;
    this->y2 = 0;
}

Linha::~Linha()
{
    delete this->aabb;
}

void Linha::geraLinha(int limite, int TamMax)
{
    float deltaX,deltaY;
    int sign_w = 0;
    int sign_h = 0;

    x1 = (rand() % limite*10)/10.0;
    y1 = (rand() % limite*10)/10.0;

    deltaX = (rand() % limite)/(float)limite;
    deltaY = (rand() % limite)/(float)limite;

    if (rand() % 2)
    {
        x2 = x1 + deltaX * TamMax;
        x2 = (x2 > 650.f) ? 650.f : x2;
    }
    else
    {
        x2 = x1 - deltaX * TamMax;
        x2 = (x2 < 0.f) ? 0.f : x2;
    }

    if (rand() % 2)
    {
        y2 = y1 + deltaY * TamMax;
        y2 = (y2 > 500.f) ? 500.f : y2;
    }
    else
    {
        y2 = y1 - deltaY * TamMax;
        y2 = (y2 < 0.f) ? 0.f : y2;
    }

    //this->aabb.reset(new AABB(x1, y1, x2, y2));
    if (this->aabb != nullptr)
        delete this->aabb;
    this->aabb = new AABB(x1, y1, x2, y2);
}


void Linha::desenhaLinha()
{
	glBegin(GL_LINES);
		glVertex2f(x1,y1);
		glVertex2f(x2,y2);
	glEnd();
}

File: Trabalho3/Linha.py
# ************************************************
#   Linha.py
#   Define a classe Linha
#   Autor: Márcio Sarroglia Pinho
#       pinho@pucrs.br
# ************************************************

from OpenGL.GL import *
from OpenGL.GLUT import *
from OpenGL.GLU import *
from Ponto import Ponto

from random import randint as rand

""" Classe Linha """
class Linha:
    def __init__(self, minx: float = 0, miny: float = 0, maxx: float = 0, maxy: float = 0, x1: float = 0, y1: float = 0, x2: float = 0, y2: float = 0):
        self.minx = minx
        self.miny = miny
        self.maxx = maxx
        self.maxy = maxy
        self.x1 = x1
        self.y1 = y1
        self.x2 = x2
        self.y2 = y2

    """ Gera uma linha com tamanho 'tamMax' dentro de um limite 'limite'. Armazena os valores nas variaveis x1,x2,y1,y2."""
    def geraLinha(self, limite: int, tamMax: int):
        self.x1 = (rand(0, limite)*10) / 10.0
        self.y1 = (rand(0, limite)*10) / 10.0

        deltaX = rand(0, limite) / limite
        deltaY = rand(0, limite) / limite

        if (rand(0, 1) % 2):
            self.x2 = self.x1 + deltaX * tamMax
        else:
            self.x2 = self.x1 - deltaX * tamMax

        if (rand(0, 2) % 2):
            self.y2 = self.y1 + deltaY * tamMax
        else:
            self.y2 = self.y1 - deltaY * tamMax

    """ Desenha a linha na tela atual """
    def desenhaLinha(self):
        glBegin(GL_LINES)
        
        glVertex2f(self.x1, self.y1)
        glVertex2f(self.x2, self.y2)

        glEnd()